query-compiler/query-compiler/src/translate/query/write.rs

```diff
@@ -187,12 +187,30 @@ pub(crate) fn translate_write_query(query: WriteQuery, builder: &dyn QueryBuilde
             record_filter,
             args,
         })) => {
+            let id_fields = model.shard_aware_primary_identifier();
+
+            let Some(selectors) = record_filter.selectors.as_ref() else {
+                if args.is_empty() {
+                    // We can just issue a read query if there are no write arguments.
+                    let query_args =
+                        QueryArguments::from((model.clone(), record_filter.filter)).with_take(Take::Some(1));
+                    let read = builder
+                        .build_get_records(&model, query_args, &id_fields, RelationLoadStrategy::Query)
+                        .map_err(TranslateError::QueryBuildFailure)?;
+                    return Ok(Expression::Unique(Expression::Query(read).into()));
+                }
```

Comment:
Why this is needed: sometimes we have update nodes that are actually used to read data.
This is semantically wrong, but that's the way it is. It's easier to handle this edge case here, than to refactor it everywhere else in the codebase in this PR.